google/cloud/internal/url_encode.cc

```diff
@@ -24,35 +25,68 @@ namespace internal {
 
 namespace {
 
-using UrlMappings =
-    std::array<std::pair<absl::string_view, absl::string_view>, 25>;
-
-constexpr auto kEncodeMappings = UrlMappings{{
-    {" ", "%20"}, {"\"", "%22"}, {"#", "%23"},  {"$", "%24"}, {"%", "%25"},
-    {"&", "%26"}, {"+", "%2B"},  {",", "%2C"},  {"/", "%2F"}, {":", "%3A"},
-    {";", "%3B"}, {"<", "%3C"},  {"=", "%3D"},  {">", "%3E"}, {"?", "%3F"},
-    {"@", "%40"}, {"[", "%5B"},  {"\\", "%5C"}, {"]", "%5D"}, {"^", "%5E"},
-    {"`", "%60"}, {"{", "%7B"},  {"|", "%7C"},  {"}", "%7D"}, {"\177", "%7F"},
-}};
-
-// Until we require C++17, it isn't worth trying to eliminate the duplication
-// using a constexpr function to produce kDecodeMappings from kEncodeMappings.
-constexpr auto kDecodeMappings = UrlMappings{{
-    {"%20", " "}, {"%22", "\""}, {"%23", "#"},  {"%24", "$"}, {"%25", "%"},
-    {"%26", "&"}, {"%2B", "+"},  {"%2C", ","},  {"%2F", "/"}, {"%3A", ":"},
-    {"%3B", ";"}, {"%3C", "<"},  {"%3D", "="},  {"%3E", ">"}, {"%3F", "?"},
-    {"%40", "@"}, {"%5B", "["},  {"%5C", "\\"}, {"%5D", "]"}, {"%5E", "^"},
-    {"%60", "`"}, {"%7B", "{"},  {"%7C", "|"},  {"%7D", "}"}, {"%7F", "\177"},
-}};
+std::string ToHex(int i) {
+  std::stringstream stream;
+  stream << "%" << std::setfill('0') << std::setw(2) << std::uppercase
+         << std::hex << i;
+  return stream.str();
+}
```

Comment:
It looks like I forgot to save one of my comments.  I think it said something like ...

Is streaming overkill for such a conversion?  For example, consider ...
```
std::string ToHex(unsigned char c) {
  std::string s = "%00";
  const char kDigits[] = "0123456789ABCDEF";
  s[1] = kDigits[(c >> 4) & 0xf];
  s[2] = kDigits[c & 0xf];
  return s;
}
```

Reply:
Ack, I think I was trying to use the STL as much as possible, but it's simpler to do your above suggestion.